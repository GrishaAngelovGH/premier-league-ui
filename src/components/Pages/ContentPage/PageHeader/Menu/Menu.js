import { Component } from 'react'
import { Link } from 'react-router-dom'

class Menu extends Component {
    state = {
        tabIndex: 0
    }

    handleTabClick = ({ target }) => {
        this.setState({ tabIndex: parseInt(target.id) })
    }

    render() {
        const { tabIndex } = this.state
        const categories = ['matches', 'news', 'standings', 'stats', 'players']

        return (
            <div className='row flex-nowrap overflow-auto'>
                <div className='col-md-12 mt-2'>
                    <div className='row flex-nowrap overflow-auto ml-1'>
                        {
                            categories.map((v, i) => {
                                const classnames = i === tabIndex ? 'border-bottom border-white item' : 'item'

                                return (
                                    <Link to={`/content/${v}`} key={i} onClick={this.handleTabClick}>
                                        <div id={i} className={classnames}>{v}</div>
                                    </Link>
                                )
                            })
                        }
                    </div>
                </div>
            </div>
        )
    }
}

export default Menu