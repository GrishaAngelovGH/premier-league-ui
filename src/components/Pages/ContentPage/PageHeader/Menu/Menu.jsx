import { useState } from 'react'
import { Link } from 'react-router-dom'

const Menu = () => {
	const [tabIndex, setTabIndex] = useState(0)

	const handleTabClick = ({ target }) => {
		setTabIndex(parseInt(target.id))
	}

	const categories = ['matches', 'news', 'standings', 'stats', 'players']

	return (
		<div className='flex gap-4 overflow-auto justify-evenly'>
			{
				categories.map((v, i) => {
					const linkClass = i === tabIndex ? 'border-b border-white hover:border-b-0' : 'border-b-0'

					return (
						<Link
							key={i}
							id={i}
							to={`/content/${v}`}
							className={`text-decoration-none text-white text-uppercase ${linkClass}`}
							onClick={handleTabClick}
						>
							{v}
						</Link>
					)
				})
			}
		</div>
	)
}

export default Menu