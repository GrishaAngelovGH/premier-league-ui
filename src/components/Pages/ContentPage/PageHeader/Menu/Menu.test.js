import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import Menu from './Menu'

test('should render Menu component', () => {
    const view = render(<Menu />, { wrapper: MemoryRouter })

    expect(view).toMatchSnapshot()
})