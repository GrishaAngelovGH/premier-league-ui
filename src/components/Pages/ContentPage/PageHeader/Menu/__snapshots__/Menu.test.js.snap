// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Menu component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row flex-nowrap overflow-auto"
      >
        <div
          class="col-md-12 mt-2"
        >
          <div
            class="row flex-nowrap overflow-auto ml-1"
          >
            <a
              class="border-bottom border-white item mx-2"
              href="/content/matches"
              id="0"
            >
              matches
            </a>
            <a
              class="item mx-2"
              href="/content/news"
              id="1"
            >
              news
            </a>
            <a
              class="item mx-2"
              href="/content/standings"
              id="2"
            >
              standings
            </a>
            <a
              class="item mx-2"
              href="/content/stats"
              id="3"
            >
              stats
            </a>
            <a
              class="item mx-2"
              href="/content/players"
              id="4"
            >
              players
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row flex-nowrap overflow-auto"
    >
      <div
        class="col-md-12 mt-2"
      >
        <div
          class="row flex-nowrap overflow-auto ml-1"
        >
          <a
            class="border-bottom border-white item mx-2"
            href="/content/matches"
            id="0"
          >
            matches
          </a>
          <a
            class="item mx-2"
            href="/content/news"
            id="1"
          >
            news
          </a>
          <a
            class="item mx-2"
            href="/content/standings"
            id="2"
          >
            standings
          </a>
          <a
            class="item mx-2"
            href="/content/stats"
            id="3"
          >
            stats
          </a>
          <a
            class="item mx-2"
            href="/content/players"
            id="4"
          >
            players
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
